pkg/orgIdColumnPopulator/main.go

```diff
@@ -0,0 +1,161 @@
+
+import (
+	"context"
+	"database/sql"
+	"fmt"
+	"log"
+	"time"
+
+	_ "github.com/lib/pq"
+
+	"github.com/RedHatInsights/tenant-utils/pkg/tenantid"
+)
+
+func MapAccountToOrgId(ctx context.Context, database *sql.DB, table, accountColumn, orgIdColumn string, dbOperationTimeout int, batchSize int, batchTranslator tenantid.BatchTranslator, recordMetrics MetricsRecorder) error {
+
+	uniqueAccounts, rowsUpdated, mappingErr := mapAccountToOrgId(ctx, database, table, accountColumn, orgIdColumn, dbOperationTimeout, batchSize, batchTranslator)
+
+	// Record the metrics even if there was an error??
+	_ = recordMetrics(table, uniqueAccounts, rowsUpdated)
+
+	return mappingErr
+}
+
+func mapAccountToOrgId(ctx context.Context, database *sql.DB, table, accountColumn, orgIdColumn string, dbOperationTimeout int, batchSize int, batchTranslator tenantid.BatchTranslator) (int64, int64, error) {
+
+	log.Println("table:", table)
+	log.Println("accountColumn:", accountColumn)
+	log.Println("orgIdColumn:", orgIdColumn)
+	log.Println("batchSize:", batchSize)
+
+	var uniqueAccounts int64
+	var rowsUpdated int64
+
+	for {
+
+		eans, err := readAccountsFromDatabase(ctx, database, table, accountColumn, orgIdColumn, batchSize, dbOperationTimeout)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		log.Printf("Read %d unique accounts from table %s\n", len(eans), table)
+
+		if len(eans) == 0 {
+			return uniqueAccounts, rowsUpdated, nil
+		}
+
+		translationResults, err := batchTranslator.EANsToOrgIDs(ctx, eans)
+		if err != nil {
+			return uniqueAccounts, rowsUpdated, err
+		}
+
+		for _, results := range translationResults {
+
+			uniqueAccounts++
+
+			if results.Err != nil {
+				log.Printf("Skipping translation of EAN (%s) to OrgID (%s) due to error: %s\n", results.EAN, results.OrgID, results.Err)
+				continue
+			}
+
+			if results.EAN == nil {
+				log.Printf("Skipping nil EAN\n")
+				continue
+			}
+
+			updatedRowsCount, err := updateOrgIdInDatabase(ctx, database, table, accountColumn, orgIdColumn, *results.EAN, results.OrgID, dbOperationTimeout)
+			if err != nil {
+				return uniqueAccounts, rowsUpdated, err
+			}
+
+			if updatedRowsCount == 0 {
+				// I can't really think of why this would happen...but I really don't want
+				// this thing to just run forever...
+				log.Println("FAILED TO UPDATED ANY ROWS...Looks like we are done!!")
+				return uniqueAccounts, rowsUpdated, fmt.Errorf("Failed to update any rows.")
+			}
+
+			rowsUpdated = rowsUpdated + updatedRowsCount
+		}
+	}
+
+	return uniqueAccounts, rowsUpdated, nil
+}
+
+func readAccountsFromDatabase(ctx context.Context, database *sql.DB, table, accountColumn, orgIdColumn string, chunkSize int, timeout int) ([]string, error) {
+
+	ctx, cancel := context.WithTimeout(ctx, time.Duration(timeout)*time.Second)
+	defer cancel()
+
+	accountList := make([]string, 0, chunkSize)
+
+	selectQuery := fmt.Sprintf("SELECT %s FROM %s WHERE %s IS NULL OR %s = '' GROUP BY %s LIMIT %d",
```

Comment:
feature request: in playbook-dispatcher we have the column defaulting to 'unknown' https://github.com/RedHatInsights/playbook-dispatcher/blob/master/migrations/008_add-org-id.up.sql Can you make the "null-placeholder" value optionally configurable?

Reply:
Great idea.  Done.